course.py

```diff
@@ -248,3 +248,44 @@ def add_duplicate_course():
         db.session.rollback()
 
     return redirect(url_for('course.course_info', course_id=course_id, current_year=year))
+
+
+@course_bp.route('/evaluations')
+@login_required
+def evaluations():
+    user_id = request.args.get('user_id')
```

Comment:
It seems you are confused between route and GET arguments. The route processor embeds regex matching and argument existence.

It is therefore more interesting to rely on the route processor for mandatory arguments and on the GET arguments for optional ones.

In this case, user id and current year should probably be part of the route arguments as you would to ensure the argument is specified and is correct.

Reply:
I understand the difference better now, thank you!